### ISSUE TITLE
Deploy with helm in custom namespace breaks k8s-hatchery

### ISSUE THREAD

**Issue by tzilist** [ovh/cds#3960]
I tried deploying CDS into a custom namespace (named cds) via helm. This is the error the k8s-hatchery service returned is this:
```
Starting service hatchery:kubernetes
Unable to init service hatchery:kubernetes: ApplyConfiguration: internal server error (caused by: Cannot create namespace cds in kubernetes: Post https://1.1.1.1:8443/api/v1/namespaces: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers))
```

I believe it is from a log issue from this section here:
https://github.com/ovh/cds/blob/master/engine/hatchery/kubernetes/kubernetes.go#L97-L105

It looks like the KUBERNETESMASTERURL is not being set properly.

**Comment by tzilist** [comment]
Update: Turns out I didn't copy my kubeconfig. Now getting this error:

```
Starting service hatchery:kubernetes
Unable to init service hatchery:kubernetes: ApplyConfiguration: internal server error (caused by: Cannot create client with newForConfig: No Auth Provider found for name "gcp")
```

**Comment by bnjjj** [comment]
As mentioned here https://github.com/ovh/cds/blob/master/contrib/helm/cds/README.md#introduction, you must use the same namespace for your hatchery and the namespace in which you deploy it. 

Then for your master url you don’t have to indicate this one if you use a kubeconfig file. Please check that this kubeconfig.yaml is good named and in the directory contrib/helm/cds

**Comment by tzilist** [comment]
@bnjjj thanks for your quick response!

It was all deployed in the same namespace. It loos like the kubeconfig.yaml does not work with GKE? Not entirely sure. This is the example kubeconfig.yaml I have, can you tell me if this looks correct?

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: xxxxxxxxxxxx
    server: https://35.236.55.74
  name: gke_xxx_zone-a_xxx
contexts:
- context:
    cluster: gke_xxx_zone-a_xxx
    user: gke_xxx_zone-a_xxx
  name: gke_xxx_zone-a_xxx
current-context: gke_xxx_zone-a_xxx
kind: Config
preferences: {}
users:
- name: gke_xxx_zone-a_xxx
  user:
    auth-provider:
      config:
        cmd-args: config config-helper --format=json
        cmd-path: /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin/gcloud
        expiry: "2019-02-24T09:42:30Z"
        expiry-key: '{.credential.token_expiry}'
        token-key: '{.credential.access_token}'
      name: gcp
```

**Comment by tzilist** [comment]
@bnjjj thanks for your quick response!

It was all deployed in the same namespace. It loos like the kubeconfig.yaml does not work with GKE? Not entirely sure. This is the example kubeconfig.yaml I have, can you tell me if this looks correct?

```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: xxxxxxxxxxxx
    server: https://ipaddress
  name: gke_xxx_zone-a_xxx
contexts:
- context:
    cluster: gke_xxx_zone-a_xxx
    user: gke_xxx_zone-a_xxx
  name: gke_xxx_zone-a_xxx
current-context: gke_xxx_zone-a_xxx
kind: Config
preferences: {}
users:
- name: gke_xxx_zone-a_xxx
  user:
    auth-provider:
      config:
        cmd-args: config config-helper --format=json
        cmd-path: /usr/local/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin/gcloud
        expiry: "2019-02-24T09:42:30Z"
        expiry-key: '{.credential.token_expiry}'
        token-key: '{.credential.access_token}'
      name: gcp
```

**Comment by tzilist** [comment]
It looks like you need to add this import to your import path to fix this issue:

```
_ "k8s.io/client-go/plugin/pkg/client/auth"
```

Reference here:

https://github.com/kubernetes/client-go/blob/master/examples/in-cluster-client-configuration/main.go#L29

**Comment by bnjjj** [comment]
Hmmm ok good catch ! You can make a pull request if you want. If not I’ll fix that asap today :)

**Comment by tzilist** [comment]
@bnjjj Done! https://github.com/ovh/cds/pull/3961

You might just want to test this out but i'm 99% sure this will work. Famous last words though :P

**Comment by bnjjj** [comment]
there is some missing vendors on your pull request and vendoring with k8s in always a pain point, so we will check ourself to include vendors with your changes. I can't fix that now but in a few hours it will be fixed. We're on it :) thank you guy.

**Comment by tzilist** [comment]
Thanks for doing that! Excited to try out CDS once it's all merged and deployed :)

**Comment by yesnault** [comment]
merged on master, this will be available on release 0.39.0
Feel free to reopen if needed
